backends/p4tools/common/compiler/convert_errors.h

```diff
@@ -0,0 +1,73 @@
+#ifndef COMMON_COMPILER_CONVERT_ERRORS_H_
+#define COMMON_COMPILER_CONVERT_ERRORS_H_
+
+#include <map>
+
+#include "frontends/common/resolveReferences/referenceMap.h"
+#include "frontends/p4/typeChecking/typeChecker.h"
+#include "frontends/p4/typeMap.h"
+#include "ir/ir.h"
+#include "lib/cstring.h"
+#include "lib/null.h"
+#include "lib/safe_vector.h"
+#include "midend/convertEnums.h"
+
+namespace P4Tools {
+
+/**
+ * Policy function: given a number of error values should return
+ * the size of a Type_Bits type used to represent the values.
+ * This class is lefted from enum conversion path.
```

Comment:
It is a modification of the convert_enums pass, there is no convert_errors pass. We were not able to consolidate those two passes, sadly. Maybe there is a good way? We can move up all these passes in the midend, since they are generally useful.